src/libos/crates/page-cache/src/cached_disk.rs

```diff
@@ -235,6 +244,87 @@ impl<A: PageAlloc> Inner<A> {
         Ok(read_len)
     }
 
+    /// Read multi pages with disk prefetch into the given buffer.
+    async fn read_multi_pages(
+        &self,
+        block_range_iter: BlockRangeIter,
+        buf: &mut [u8],
+    ) -> Result<usize> {
+        let mut read_len = 0;
+
+        // Prefetch data from disk
+        let prefetch_buf = {
```

Comment:
This prefetch is problematic in terms of both performance and correctness.

First of all, if the target pages are already cached, then the prefetch is a complete waste of time. So under certain workloads, this prefetch logic hurts performance instead of improving it.

Second, the prefetch logic may load _stale_ data into the cache. This is because after the prefetch loads the data of a block from the disk, the block may be updated by concurrent writers. Such changes may not be captured by reading the page status later. (I think you can figure out such an example)

Reply:
I refactored this part of code. Thanks for taking time to review again.
I also improve the "group_by" part in `async_sfs`, `cached_disk` and `jindisk` to calculate maximum batch length to avoid repeated and excessive memory allocation.